pkg/pillar/agentlog/agentlog.go

```diff
@@ -317,10 +330,67 @@ func roundToMb(b uint64) uint64 {
 	return mb
 }
 
+func touchFile(filename string) {
+	_, err := os.Stat(filename)
+	if os.IsNotExist(err) {
+		file, err := os.OpenFile(filename, os.O_WRONLY|os.O_CREATE|os.O_SYNC, 0755)
+		if err != nil {
+			log.Fatalf("touchFile: Failed touching file %s with err: %s", filename, err)
+		}
+		defer file.Close()
+	} else {
+		currentTime := time.Now()
+		err = os.Chtimes(filename, currentTime, currentTime)
+		if err != nil {
+			log.Fatalf("touchFile: Failed touching file %s with err: %s", filename, err)
+		}
+	}
+}
+
+// Send application stdout/stderr to this panic file
+func spoofStdFDs(agentName string) *os.File {
+	//filename := fmt.Sprintf("%s/%s", types.PersistDebugDir, panicFileName)
+	agentDebugDir := fmt.Sprintf("%s/%s/", types.PersistDebugDir, agentName)
+	if _, err := os.Stat(agentDebugDir); os.IsNotExist(err) {
+		// Create the agent specific trace directory
+		if err := os.MkdirAll(agentDebugDir, 0755); err != nil {
+			log.Fatalf("spoofStdFDs: Agent specific debug directory (%s) does not exist: %s",
+				agentDebugDir, err)
+		}
+	}
+	startTimeFile := agentDebugDir + "/starttime"
+	touchFile(startTimeFile)
+
+	panicOutFile := agentDebugDir + "/stdout"
```

Comment:
Since this no longer gets panics, a different name for the local variable would be helpful.

Reply:
done